File: src/frontend/src/app/core/models/auth.models.ts
export interface AuthResponse {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiresAt: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface RegisterRequest {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
}

export interface UserProfile {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  roles: string[];
}

export interface StoredAuthSession {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiresAt: string;
}

File: src/backend/src/Modules/Identity/Ecommerce.Modules.Identity.Application/Models/AuthModels.cs
namespace Ecommerce.Modules.Identity.Application.Models;

/// <summary>Par de tokens retornado nos fluxos de autenticação.</summary>
public sealed record AuthTokensResponse(
    string AccessToken,
    string RefreshToken,
    DateTime AccessTokenExpiresAt);

/// <summary>Access token JWT gerado internamente.</summary>
public sealed record TokenPair(string AccessToken, DateTime ExpiresAt);

/// <summary>Perfil público do usuário autenticado.</summary>
public sealed record UserProfileResponse(
    Guid Id,
    string Email,
    string FirstName,
    string LastName,
    IReadOnlyCollection<string> Roles);
